### DISCUSSION TITLE
Can´t pair using PC or Android

### DISCUSSION THREAD

**Discussion by fsikeda** [darthcloud/BlueRetro#206] [Q&A] [answered]
Hello,

I built a adapter using a devkit v4. I can pair a ipega PG-9078 controller but can´t pair if I use Windows 10 PC or Android (tried Galaxy S10, Moto G4 and Tablet Lenovo). Tried factory reset, not working.
Below I attached a serial log when I try to pair using Windows:

 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
€€ € €€€ €   € €€€€€ €€     €         €  €    €      €    €   €  €  € €€  € €€  €   €€ € €€ €    €€     €  €€€ €€€€€€€ €  €€€ €  €€€€  €€€€ €€ €€€€ € €€€€ €  €   €€€€€€€ €€ €    €€  €     € € €€ €€€  €€€€  €€ €€ €€ €€€€€€  €  €€ €€ €€  €€€€  €€ €€ € €€€€ € €€  € € €€€€ €€  €€ €€€ €€ €€ €€ € €      €   € €€ €€€  €€€€€ €€ €    €€€ €€€ €€€  € € € €€€€€€€€€  €  € €€ €    €     € € €€€€  €  € €€ €    €     €   €€€€  €  € €€ €    €     € € €€ €  €€€  €  € €€ €    €     €  €€  €€€€  €  € €€ €    €     € €€€ €€€€  €  € €€ €    €    €€  € €€€€€€€  €€€ € €   €€  €€ € €   € € €€€€€€ €€€€€€ € €€€ €   € €€€€€ €€     €         €  €    €      €    €   €  €  € €€  € €€  €   €€ € €€ €    €€     €  €€€ €€€€€€€ €  €€€ €  €€€€  €€€€ €€ €€€€ € €€€€ €  €   €€€€€€€ €€ €    €€  €     € € €€ €€€  €€€€  €€ €€ €€ €€€€€€  €  €€ €€ €€  €€€€  €€ €€ € €€€€ € €€  € € €€€€ €€  €€ €€€ €€ €€ €€ € €      €   € €€ €€€  €€€€€ €€ €    €€€ €€€ €€€  € € € €€€€€€€€€  €  € €€ €    €     € € €€€€  €  € €€ €    €     €   €€€€  €  € €€ €    €     € € €€ €  €€€  €  € €€ €    €     €  €€  €€€€  €  € €€ €    €     € €€€ €€€€  €  € €€ €    €    €€  € €€€€€€€  €€€ € €   €€  €€ € €   € € €€€€€€ €€€€€    €€ €€ €€ €    €€  €  €€€€€€ €  € €    €€    €  €  €    €     €  €€€€ € €€ € €€€  €  €  € €€  €  €€€€€€ €  € €    € €     €€          €  €€€€ € €€ € €   €        €€  €  €€€€€€ €  € €    € €          €      €  €€€€ € €€ € €      €   €€  € €€ € €€ €€  €€ €     € €          €     €€  € [0;32mI (243) cpu_start: Pro cpu up.[0m
[0;32mI (243) cpu_start: Single core mode[0m
[0;32mI (251) cpu_start: Pro cpu start user code[0m
[0;32mI (251) cpu_start: cpu freq: 240000000 Hz[0m
[0;32mI (251) cpu_start: Application information:[0m
[0;32mI (251) cpu_start: Project name:     BlueRetro[0m
[0;32mI (252) cpu_start: App version:      v1.3 universal_spiffs[0m
[0;32mI (252) cpu_start: Compile time:     Feb 26 2022 21:02:12[0m
[0;32mI (253) cpu_start: ELF file SHA256:  d64dff37bcde87bf...[0m
[0;32mI (254) cpu_start: ESP-IDF:          v5.0-dev-1426-ge899edd793[0m
[0;32mI (255) cpu_start: App cpu up.[0m
[0;32mI (255) heap_init: Initializing. RAM available for dynamic allocation:[0m
[0;32mI (256) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM[0m
[0;32mI (257) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM[0m
[0;32mI (258) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM[0m
[0;32mI (258) heap_init: At 3FFD0288 len 0000FD78 (63 KiB): DRAM[0m
[0;32mI (259) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM[0m
[0;32mI (260) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM[0m
[0;32mI (261) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM[0m
[0;32mI (262) heap_init: At 40078000 len 00008000 (32 KiB): IRAM[0m
[0;32mI (262) heap_init: At 4009E500 len 00001B00 (6 KiB): IRAM[0m
[0;32mI (263) heap_init: At 3FF80000 len 00002000 (8 KiB): RTCRAM[0m
[0;32mI (265) spi_flash: detected chip: generic[0m
[0;32mI (266) spi_flash: flash io: dio[0m
[0;32mI (266) esp_core_dump_uart: Init core dump to UART[0m
[0;32mI (266) esp_core_dump_uart: Init core dump to UART[0m
[0;32mI (267) cpu_start: Starting scheduler on PRO CPU.[0m
 bt_host_load_bdaddr_from_file: No BDADDR on SD. Using ESP32's MAC
[0;32mI (377) BTDM_INIT: BT controller compile version [33445ec][0m
[0;32mI (387) system_api: Base MAC address is not set[0m
[0;32mI (387) system_api: read default base MAC address from EFUSE[0m
[0;32mI (387) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07[0m
 bt_hci_cmd_reset
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_features
[0;32mI (787) gpio: GPIO[0]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0 [0m
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_version_info
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_bd_addr
 BT_HCI_EVT_CMD_COMPLETE
 local_bdaddr: C4:5B:BE:3F:54:FA
 bt_hci_cmd_read_buffer_size
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_class_of_device
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_name
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_voice_setting
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_num_supported_iac
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_current_iac_lap
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_filter
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_conn_accept_timeout
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_supported_commands
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_ssp_mode
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_inquiry_mode
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_inquiry_rsp_tx_pwr_lvl
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_local_ext_features
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_stored_link_key
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_page_scan_activity
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_read_page_scan_type
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_le_host_supp
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_read_wl_size
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_clear_wl
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_delete_stored_link_key
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_class_of_device
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_local_name
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_filter
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_filter
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_auth_enable
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_set_event_mask
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_page_scan_activity
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_inquiry_scan_activity
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_page_scan_type
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_page_scan_timeout
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_hold_mode_act
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_scan_enable
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_write_default_link_policy
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_read_buffer_size
 BT_HCI_EVT_CMD_COMPLETE
 bt_att_set_le_max_mtu 247
 bt_hci_cmd_le_set_adv_param
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_adv_data
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_scan_rsp_data
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_adv_enable
 BT_HCI_EVT_CMD_COMPLETE
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_periodic_inquiry
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x200C error: 0x0C retry: 0
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_periodic_inquiry
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x0403 error: 0x0C retry: 0
 mc_restore: restore sucessful!
App main at 40083050
App init at 40083060
APP_CPU RESET: 0
APP_CPU CLKGATE: 0
APP_CPU STALL: 0
APP_CACHE_CTRL: 00000030
APP CPU STACK PTR: 3FFDFAD4
GPIO[19]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[21]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[22]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[25]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[32]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[33]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[34]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[35]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:3
GPIO[39]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[19]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[21]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[22]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[25]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[32]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[33]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[34]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[35]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
GPIO[39]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
 Detected system : 17: PS2
GPIO[2]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
GPIO[34]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:1
GPIO[5]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:1
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_LE_CONN_COMPLETE
 BT_ATT_OP_MTU_REQ
 Set mtu 247
 bt_att_cmd_mtu_rsp
 bt_host_acl_hdlr dev NULL!
 BT_ATT_OP_FIND_TYPE_REQ
 bt_att_cmd_error_rsp
 BT_ATT_OP_FIND_TYPE_REQ
 bt_att_cmd_error_rsp
 BT_ATT_OP_FIND_TYPE_REQ
 bt_att_cmd_error_rsp
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_DISCONN_COMPLETE
 DISCONN from BLE config interface
 bt_hci_cmd_le_set_adv_enable
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE

**Comment by darthcloud** [comment]
What version of Google Chrome are you using on Win 10?

When you see the BlueRetro device in the connection list, is the signal strength indicator over half?

**Comment by fsikeda** [comment]
Thanks for answer me. I realized after that the procedure is simply open web page and connect/pair clicking the button. I was trying to pair via Windows /android control panel before it. So I learned how to do it now.
But now I have another issue: I was using universal FW 1.3. Since I have one single PSX cable, I made single port wiring. I followed the tutorial:

If using an universal FW, make sure to connect I39 to GND and I35 to GND.
If using an universal FW AND not wiring 2nd port plug, make sure to connect IO22 & IO25 to GND.
If not wiring 2nd port plug, make sure to connect IO5 (P2_DTR), IO26 (P2_SCK) & IO27 (P2_TXD) to 3.3V

So I connected pins I39, I35, IO22, IO25 to devkit GND and pins IO5, IO26, IO27 to devkit 3V3.

I tested with my PS2 FAT console 35xxx and nothing happens. Tested all wires using multimeter and are all fine.
Upgraded to universal FW 1.4 and stills same (BT controller pairing OK, but no commands to PS2 console.
Is there a way to test if buttons are OK? I didn´t understand how testjig works.

**Reply by darthcloud** [reply]
take some pictures where I can see the connection easily.

**Comment by fsikeda** [comment, marked as answer]
Sorry for late reply. I was evolved in a accident and I hurted my right hand last afternoon. So my hand is inactive for some days.
Before it, I could solve the problem. I´ve checked all wires and all ok. I have a ancient PS2 -> USB adapter for PC. I tried test Blueretro connecting into this adapter and it worked. But buttons behavior are like someone are pressing them randonly.
I´ve read all discutions about PS related and I´ve found that cable length has influence on it. I was using a cable from a defective controller and it still had original length. So I cutted to 1 feet length and now it worked connected to PS2 console.
I´ve ordered a generic cheap controller that will be the cable donor for 2nd player. Until it doesn´t come I will recover my hand in that time and so I can finish assembly and take some pictures.

Another odd behavior I noticed is: first I flashed universal 1.3 spiffs. After many tests before I solve the problem, I updated to 1.4 through OTA web app. It was recognized with new FW 1.4. But after this, I´ve messed buttons config and I decided to do a factory reset (10s button press). After factory reset, FW was reverted to 1.3. I´ve tried couple times doing the same way (update via web app and do factory rst) and the behaviour was the same (new FW is reverted to previous FW). I solved only when I re flashed via esptools.

Thanks for attention and for this amazing project!

**Reply by darthcloud** [reply]
Yes factory reset the firmware to the originally flashed one.

**Reply by fsikeda** [reply]
Ok, understood. So, is it possible keep 2 fw? Maybe it could be useful to maintenance/test, keep testpad along other fw, selecting via serial or keep some button pressed at boot.
